File: fsharp/anagram/Anagram.fs
module Anagram

open System

let findAnagrams (sources: List<string>) (target: string) : List<string> =
    let normalize string =
        string |> Seq.map Char.ToLower |> Seq.sort |> Seq.toList

    sources
    |> List.filter (fun w -> w.ToLower() <> target.ToLower())
    |> List.filter (fun w -> normalize w = normalize target)

File: haskell/anagram/src/Anagram.hs
module Anagram (anagramsFor) where

import Data.Char (toLower)
import Data.Function
import Data.List (sort)

anagramsFor :: String -> [String] -> [String]
anagramsFor xs xss =
  filter (\x -> map toLower x /= map toLower xs) xss
    & filter (\x -> sort (map toLower x) == sort (map toLower xs))
